react-validated-input-component/my-app/src/ValidatedInput.jsx

```diff
@@ -0,0 +1,38 @@
+import './styles.css'
+import { useState } from 'react'
+
+export default function ValidatedInput() {
+  const [checkMark, setCheckMark] = useState(false);
+  const [message, setMessage] = useState('');
+
+  function handleChange(e) {
+    if (e.target.value === '') {
+      setCheckMark(false);
+      setMessage('A password is requried')
+    } else if (e.target.value.length < 8) {
+      setCheckMark(false);
+      setMessage('Your password is too short');
+    } else if (!/[0-9]/.test(e.target.value)) {
+      setCheckMark(false);
+      setMessage('Your password needs to contain at least one number');
+    } else if (!/[A-Z]/.test(e.target.value)) {
+      setCheckMark(false);
+      setMessage('Your password needs to contain at least one capital letter');
+    } else if (!/[!@#$%^&*()]/.test(e.target.value)) {
+      setCheckMark(false);
+      setMessage('Your password needs to contain at least one special character: !, @, #, $, %, ^, &, *, (, or )');
+    } else {
+      setCheckMark(true);
+      setMessage(' ');
+    }
+  }
+
+  return (
+   <div className = "form-group">
+      <label htmlFor="inputPassword5" className="form-label col-sm-12 ml-2" id="password-label">Password</label>
+      <input type="password" id="inputPassword5" className="form-control w-50" aria-labelledby="passwordHelpBlock" onChange={handleChange} />
```

Comment:
@wang-lawrence While I like that you added more realistic password constraints, in practice it's more useful and robust if the form values themselves are stored in state, not other displayed content & styling that can be directly calculated based off of them.